qa/integration-tests/src/test/java/io/zeebe/broker/it/clustering/SingleBrokerDataDeletionTest.java

```diff
@@ -58,6 +67,100 @@ private void configureCustomExporter(final BrokerCfg brokerCfg) {
     brokerCfg.setExporters(Collections.singletonMap("snapshot-test-exporter", exporterCfg));
   }
 
+  @Test
+  public void shouldCompactEvenIfSkippingAllRecords() {
+    // given
+    final Broker broker = clusteringRule.getBroker(0);
+
+    // when
+    ControllableExporter.updatePosition(false);
+    ControllableExporter.RECORD_TYPE_FILTER.set(r -> r == RecordType.COMMAND);
+    ControllableExporter.VALUE_TYPE_FILTER.set(r -> r == ValueType.DEPLOYMENT);
+    writeSegments(broker, 2);
+    clusteringRule
+        .getClient()
+        .newDeployCommand()
+        .addWorkflowModel(
+            Bpmn.createExecutableProcess("process").startEvent().done(), "process.bpmn")
+        .send()
+        .join();
+    await("until at least one record is exported")
+        .atMost(Duration.ofSeconds(5))
+        .untilAsserted(
+            () -> assertThat(ControllableExporter.EXPORTED_RECORDS).hasValueGreaterThan(0));
+
+    // enforce compaction
+    final var segmentsBeforeSnapshot = getSegmentsCount(broker);
+    clusteringRule.getClock().addTime(SNAPSHOT_PERIOD);
+
+    // then
+    assertThat(clusteringRule.waitForSnapshotAtBroker(broker)).isNotNull();
+    await()
+        .untilAsserted(
+            () ->
+                assertThat(getSegmentsCount(broker))
+                    .describedAs("Expected less segments after a snapshot is taken")
+                    .isLessThan(segmentsBeforeSnapshot));
+  }
+
+  @Test
+  public void shouldNotCompactUnacknowledgedEventsEvenIfSkipping() {
```

Comment:
These IT tests look complicated and just not right :see_no_evil: 
However, I don't have a better idea for now.

Reply:
Agreed, and they know too much implementation details (e.g. where to find snapshots, how to count segments, etc.) :sweat_smile: I'll give another hour to see if I can't come up with something else

Reply:
I think I have a concept which could help make these tests better - it's kind of two parter. I have an idea how to make tests more maintainable and less coupled to implementation details, but that requires more refactoring. In the meantime I have another idea to make it more maintainable with a bit less refactoring, and I'll explore that in a different PR to avoid the burden here.